### ISSUE TITLE
Does the open source model support multi-view video generation from horizontally concatenated multi-view images with a shared prompt?

### ISSUE THREAD

**Issue by supeng0924** [open-gigaai/giga-world-0#8]
I'm wondering whether the current open-source video generation model supports the following scenario:

Input: A single image that consists of multiple views of the same object or scene, concatenated horizontally (e.g., left view | center view | right view).
Prompt: A text prompt describing the desired motion or content.
Output: A temporally coherent video where multi-view consistency is preserved across frames (i.e., the generated video respects the spatial relationships implied by the multi-view input).

In a robotics scenario, I tried horizontally concatenating three images with aligned timestamps as input, but the model's output was very poor. I'm not sure whether this is due to incorrect usage on my part or if the model simply doesn't support this setup.

**Comment by 1zeryu** [comment]
Thank you for your insightful question!

Currently, the released open-source video generation model does not support generating temporally coherent videos that preserve multi-view spatial consistency from horizontally concatenated multi-view images as input. The approach you described—directly concatenating multiple views at the pixel level—typically cannot yield satisfactory multi-view coherence in the output.

Internally, for multi-view scenarios, our recent experiments take a different approach:
Instead of pixel-level concatenation, we encode each view separately using a VAE. The resulting latent representations from the three views are then concatenated along the spatial dimension in latent space. In our comparisons, training with this method produces much better results compared to pixel-space concatenation or concatenation along the temporal dimension in latent space.

**Comment by supeng0924** [comment]
Thank you for your answer.
Based on previous replies, I would like to confirm the specific implementation logic to avoid any misunderstanding.
I noticed that in pipeline_giga_world_0.py, line 233 is cond_latents = self.encode(cond_images).to(dtype).
Does the official suggestion of "concatenating along the spatial dimension in the latent space" mean we need to modify the pipeline to encode each of the three views separately with the VAE first, and then concatenate them?
Example for Confirmation:
Assuming the input resolution for a single view is 480x640, and the corresponding Latent temporal dimension for the generated video is 16.
Is the workflow as follows:
Encode Separately:
cond_latents_left = self.encode(cond_images_left) -> Shape: (1, 16, 16, 60, 80)
cond_latents_mid = self.encode(cond_images_mid) -> Shape: (1, 16, 16, 60, 80)
cond_latents_right = self.encode(cond_images_right) -> Shape: (1, 16, 16, 60, 80)
(Note: Shape format is B, C, T, H, W, where C=16)
Spatial Concatenation:
Concatenate along the width dimension (the last dimension):
cond_latents = torch.cat([cond_latents_left, cond_latents_mid, cond_latents_right], dim=-1)
Final Shape: (1, 16, 16, 60, 240)
Input to Model:
Feed this widened cond_latents as a condition into the DiT for noise prediction.
Core Questions:
Is the above understanding completely correct?
Does the currently released open-source model weight (in the Inference stage) directly support this kind of input?
Since the width of the concatenated Latent becomes 3 times the original (W=240), can the model's Positional Embedding generalize to handle this unseen aspect ratio?
Or does this method require fine-tuning on specific multi-view data to be effective?

**Comment by 1zeryu** [comment]
Your workflow is correct, but effective generation using this method requires either fine-tuning the model or using weights trained with similar multi-view concatenated latent inputs.
